libs/designsystem/src/lib/components/radio/radio-group/radio-group.component.ts

```diff
@@ -121,44 +132,98 @@ export class RadioGroupComponent implements AfterContentInit, AfterViewInit {
+    let selectedIndex = -1;
+    this.projectedRadioButtons &&
+      this.projectedRadioButtons.find((radio, i) => {
+        const found = radio.value === value;
+        if (found) {
+          selectedIndex = i;
+        }
+        return found;
+      });
+    return selectedIndex;
+  }
+
+  private getValueFromSelectedIndex(): string | any {
+    if (this.selectedIndex === -1) return;
+    return this.hasItemsFromContentProjection
+      ? this.getValueFromProjectedRadios(this.selectedIndex)
+      : this.items[this.selectedIndex];
+  }
+
+  private getValueFromProjectedRadios(index: number): string | any {
+    const radio =
+      this.projectedRadioButtons &&
+      this.projectedRadioButtons.find((_, i) => {
+        return i === index;
+      });
+    return radio && radio.value;
+  }
+
+  private initSelectionStateFromProjectedContent() {
+    if (this.hasItemsFromContentProjection) {
+      this.refreshSelectionState(); // Initialize selected index and value from projected radios
     }
-    const selectedRadio =
-      this.slottedRadioButtons && this.slottedRadioButtons.toArray()[this.selectedIndex];
-    return selectedRadio && selectedRadio.value; // Get value from slotted radios
   }
 
-  private setDisabledNextCycle() {
-    if (this.disabled) {
-      setTimeout(() => this.setDisabledState(this.disabled));
-    }
-  }
+  private listenForProjectedRadiosChange() {
+    this.projectedRadioButtons.changes.subscribe(() => {
+      this.refreshStateFromProjectedContent();
 
-  private setRadioDisabledState(radioButtons: QueryList<RadioComponent>, isDisabled: boolean) {
-    if (!radioButtons || !radioButtons.length) return;
-    radioButtons.forEach((radio, index) => {
-      let disableRadio = isDisabled;
-      if (isDisabled === false) {
-        disableRadio = this._getDisabledStateFromItem(this.items[index]); // Ensure each item's disabled state overwrite radio-group state, if defined
+      // Ensure disabled state propagates when re-rendering projected radios.
+      // setTimeout prevents ExpressionChangedAfterItHasBeenCheckedError when updating the DOM in QueryList.changes:
+      if (this.disabled) {
+        setTimeout(() => this.setProjectedRadiosDisabledState(this.disabled));
       }
-      radio.disabled = disableRadio;
     });
   }
 
-  private setSelectedItem(value: any) {
-    if (value === this._value) return;
+  private refreshSelectionState() {
+    if (this.value) {
+      this._selectedIndex = this.getIndexOfSelectedValue(); // Ensure selectedIndex reflects value within items
+    }
+    this._value = this.getValueFromSelectedIndex() || null; // Ensure value exists within items
+  }
+
+  private refreshStateFromProjectedContent() {
+    if (!!this._customItemTemplate) return; // Only refresh on changes to projected content, not when re-rendering custom template
+    this.changeDetectionRef.markForCheck(); // Ensure changes to projected content triggers change detection
```

Comment:
Mark for check doesn't trigger change detection but ensures the component gets checked on next CD/tick

Reply:
@lydemann Good point ;)
Updated the comment 👍